pkg/prometheus/resource_selector.go

```diff
@@ -442,6 +483,13 @@ func (rs *ResourceSelector) SelectPodMonitors(ctx context.Context, listFn ListAl
 				continue
 			}
 
+			if endpoint.OAuth2 != nil {
+				if err = validateProxyConfig(ctx, endpoint.OAuth2.ProxyConfig, rs.store, pm.GetNamespace()); err != nil {
```

Comment:
we should rather have a validateOAuth2() function to validate all the fields and call store.AddOAuth2() which would only deal with retrieving the secret key refs.